### PR TITLE
Fxpool oracle decimals

### PR DESCRIPTION
**Pull request by schystz**
# Description

Adds a new property `fxOracleDecimals` to `Token` entity, and use this new property (instead of a hardcoded decimal) when computing for `latestFXPrice`.

### Motivation

We are updating support of FXPools in the  balancer-sor and our initial changes has been already merged on [balancer/balancer-sor#305](https://github.com/balancer/balancer-sor/pull/305). In this update, we are adding improvements on the accuracy of the returned values and fix the bug that was found by the sor development team (see https://github.com/balancer/balancer-sor/tree/fx-pool-debug).

`fxOracleDecimals` will primarily be used by `fxpoolMath` to ensure that all values are in wei from when we calculate things in the SOR. This will also include support of using our bb-usd oracles for the next version of our FxPool that we will be calling FxMetaPool.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

We've already deployed the changes on our subgraph clone here - https://thegraph.com/hosted-service/subgraph/xave-finance/balancer-v2

Simply select "fxOracleDecimals" from the dropdown and run it to see the results.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 1 review (If the PR is significant enough, use best judgement here)
- [x] I have commented my code where relevant, particularly in hard-to-understand areas

### `dev` -> `master`
- [ ] I have [checked](https://balancer.github.io/balancer-subgraph-v2/status.html) that all beta deployments have synced
- [ ] I have [checked](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-polygon-prune-v2-beta/graphql?query=%0A%7B%0A++balancers%28block%3A%7Bnumber%3A1%7D%29%7B%0A++++id%0A++%7D%0A%7D) that the earliest block in the polygon pruned deployment is [block, date/time](https://polygonscan.com/block/block)
  - [ ] The earliest block is more than 24 hours old
- [ ] I have checked that core metrics are the same in the beta and production deployments

### Merges to `dev`
- [ ] I have checked that the graft base is not a pruned deployment

### CODE REVIEW

File: src/mappings/helpers/misc.ts [hunk 1]
```diff
@@ -275,6 +285,14 @@ export function createToken(tokenAddress: Address): Token {
     token.pool = poolId.toHexString();
   }
 
+  // attempt to retrieve oracle decimals for FX tokens
```
Comment (mendesfabio) [inline src/mappings/helpers/misc.ts:288]: the PR looks good, but I'm wondering if makes sense to try to get the decimals on-chain. The Oracles you've added all have 8 decimals (and that seems to be a pattern for non-ETH pairs). If you want to support a new asset/oracle n the future you'll have to redeploy the Subgraph anyway, right? Maybe hardcoding directly the decimals (if the token is in FX_TOKEN_ADDRESSES) would be easier in that case - or if there's a way to go from Aggregator -> Oracle -> Decimals that would be perfect.

Comment (mendesfabio) [reply src/mappings/helpers/misc.ts:288]: >  if there's a way to go from Aggregator -> Oracle -> Decimals that would be perfect.
I assume this isn't possible, right @schystz?

Comment (schystz) [reply src/mappings/helpers/misc.ts:288]: Hi @mendesfabio , yeah it looks like we can get the [aggregator from the oracle](https://polygonscan.com/address/0x8CE3cAc0E6635ce04783709ca3CC4F5fc5304299#readContract#F2) but not the other way around.

> If you want to support a new asset/oracle n the future you'll have to redeploy the Subgraph anyway, right?

We just wanna be extra careful and make the code future proof, but we can totally hardcode it until we add the oracles with decimals other than `8`.

Comment (schystz) [reply src/mappings/helpers/misc.ts]: With the fix above, the PR changes has considerably went down. Can you please check now @mendesfabio ?

**Review by mendesfabio — APPROVED** [review]

**Comment by mendesfabio** [comment]
I'm deploying this to staging right now -- let you know once it's synced! 🙏

**Pull request merged**